src/axom/quest/detail/MarchingCubesFullParallel.hpp

```diff
@@ -0,0 +1,897 @@
+      int index = 0;
+      for(int n = 0; n < CELL_CORNER_COUNT; ++n)
+      {
+        if(f[n] >= contourVal)
+        {
+          const int bit = (1 << n);
+          index |= bit;
+        }
+      }
+      return index;
+    }
+
+    template <int TDIM = DIM>
+    AXOM_HOST_DEVICE inline typename std::enable_if<TDIM == 2>::type
+    computeCaseId(axom::IndexType i, axom::IndexType j) const
+    {
+      const bool useZone = maskView.empty() || bool(maskView(i, j));
+      if(useZone)
+      {
+        // clang-format off
+          double nodalValues[CELL_CORNER_COUNT] =
+            {fcnView(i    , j    ),
+             fcnView(i + 1, j    ),
+             fcnView(i + 1, j + 1),
+             fcnView(i    , j + 1)};
+        // clang-format on
+        caseIdsView(i, j) = computeCrossingCase(nodalValues);
+      }
+    }
+
+    //!@brief Populate m_caseIds with crossing indices.
+    template <int TDIM = DIM>
+    AXOM_HOST_DEVICE inline typename std::enable_if<TDIM == 3>::type
+    computeCaseId(axom::IndexType i, axom::IndexType j, axom::IndexType k) const
+    {
+      const bool useZone = maskView.empty() || bool(maskView(i, j, k));
+      if(useZone)
+      {
+        // clang-format off
+          double nodalValues[CELL_CORNER_COUNT] =
+            {fcnView(i + 1, j    , k    ),
+             fcnView(i + 1, j + 1, k    ),
+             fcnView(i    , j + 1, k    ),
+             fcnView(i    , j    , k    ),
+             fcnView(i + 1, j    , k + 1),
+             fcnView(i + 1, j + 1, k + 1),
+             fcnView(i    , j + 1, k + 1),
+             fcnView(i    , j    , k + 1)};
+        // clang-format on
+        caseIdsView(i, j, k) = computeCrossingCase(nodalValues);
+      }
+    }
+  };  // MarkCrossings_Util
+
+  /*!
+    @brief Populate the 1D m_contourNodeCoords, m_contourCellCorners
+    and m_contourCellParents arrays that defines the unstructured
+    contour mesh.
+  */
+  void scanCrossings()
+  {
+  #ifdef __INTEL_LLVM_COMPILER
+    // Intel oneAPI compiler segfaults with OpenMP RAJA scan
+    using ScanPolicy =
+      typename axom::execution_space<axom::SEQ_EXEC>::loop_policy;
+  #else
+    using ScanPolicy = typename axom::execution_space<ExecSpace>::loop_policy;
+  #endif
+
+    const axom::IndexType parentCellCount = m_caseIds.size();
+    auto caseIdsView = m_caseIds.view();
+
+    // Compute number of surface facets added by each parent cell.
+    m_facetIncrs.resize(parentCellCount);
+    const auto facetIncrsView = m_facetIncrs.view();
+
+  #if defined(AXOM_USE_RAJA)
+    axom::for_all<ExecSpace>(
+      0,
+      parentCellCount,
+      AXOM_LAMBDA(axom::IndexType parentCellId) {
+        facetIncrsView.flatIndex(parentCellId) =
+          num_contour_cells(caseIdsView.flatIndex(parentCellId));
+      });
+  #else
+    for(axom::IndexType pcId = 0; pcId < parentCellCount; ++pcId)
+    {
+      facetIncrsView.flatIndex(pcId) =
+        num_contour_cells(caseIdsView.flatIndex(pcId));
+    }
+  #endif
```

Comment:
@gunney1 -- since there are separate implementations for "FullParallel" and "PartialParallel" would it make sense to have a separate implementation for pure sequential -- i.e. when we configure without RAJA? If so, that might clean up the code significantly.

Reply:
I think that we discussed implementing sequential implementations for use when configured without RAJA.  Something like we have in https://github.com/LLNL/axom/blob/39381a84ddd9de2f4538b4d248917810ebb30ace/src/axom/mint/execution/internal/for_all_nodes.hpp#L53  When we have that, we can remove the non-RAJA `#else` blocks, here and elsewhere in Axom.

Reply:
Update: I was referring to multidimensional nested loops.  For non-nested loops, the current `for_all` should work.  I will remove the `#if/#else` for them.